jgrapht-core/src/main/java/org/jgrapht/intervalgraph/IntervalGraph.java

```diff
@@ -1,14 +1,468 @@
+     * an edge that its source and target vertices are the same.
+     *
+     * @return <code>true</code> if and only if graph loops are allowed.
+     */
+    public boolean isAllowingLoops()
+    {
+        return allowingLoops;
+    }
+
+    /**
+     * Returns <code>true</code> if and only if multiple (parallel) edges are allowed in this graph. The
+     * meaning of multiple edges is that there can be many edges going from vertex v1 to vertex v2.
+     *
+     * @return <code>true</code> if and only if multiple (parallel) edges are allowed.
+     */
+    public boolean isAllowingMultipleEdges()
+    {
+        return allowingMultipleEdges;
+    }
+
+    /**
+     * Returns <code>true</code> if and only if the graph supports edge weights.
+     *
+     * @return <code>true</code> if the graph supports edge weights, <code>false</code> otherwise.
+     */
+    public boolean isWeighted()
+    {
+        return weighted;
+    }
+
+    /**
+     * Returns <code>true</code> if the graph is directed, false if undirected.
+     *
+     * @return <code>true</code> if the graph is directed, false if undirected.
+     */
+    public boolean isDirected()
+    {
+        return directed;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public E getEdge(V sourceVertex, V targetVertex)
+    {
+        return specifics.getEdge(sourceVertex, targetVertex);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public EdgeFactory<V, E> getEdgeFactory()
+    {
+        return edgeFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public E addEdge(V sourceVertex, V targetVertex)
+    {
+        throw new IllegalArgumentException(INTERVAL_GRAPH_ADD_EDGE);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean addEdge(V sourceVertex, V targetVertex, E e)
+    {
+        throw new IllegalArgumentException(INTERVAL_GRAPH_ADD_EDGE);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean addVertex(V v)
+    {
+        if (v == null) {
+            throw new NullPointerException();
+        } else if (containsVertex(v)) {
+            return false;
+        } else {
+            specifics.addVertex(v);
+
+            return true;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V getEdgeSource(E e)
+    {
+        return intrusiveEdgesSpecifics.getEdgeSource(e);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V getEdgeTarget(E e)
+    {
+        return intrusiveEdgesSpecifics.getEdgeTarget(e);
+    }
+
+    /**
+     * Returns a shallow copy of this graph instance. Neither edges nor vertices are cloned.
+     *
+     * @return a shallow copy of this graph.
+     *
+     * @throws RuntimeException in case the clone is not supported
+     *
+     * @see java.lang.Object#clone()
+     */
+    @Override
+    public Object clone()
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/lang/Cloneable.html according to the language specification, clone indicates field-to-field copy, namely deep copy, so it may not be appropriate to implement clone method and return shallow copy instead. Consider providing additional constructor method if needed, then the class does not need to be Cloneable.

Reply:
Other graph implementations do it likes this, too.